lib/vicadmin/validate.go

```diff
@@ -71,46 +73,64 @@ func NewValidator(ctx context.Context, vch *config.VirtualContainerHostConfigSpe
 	}
 	log.Infof("Setting version to %s", v.Version)
 
-	//VCH Name
+	// VCH Name
 	v.Hostname, _ = os.Hostname()
 	v.Hostname = strings.Title(v.Hostname)
 	log.Infof("Setting hostname to %s", v.Hostname)
 
-	//Firewall Status Check
-	v2, _ := validate.CreateFromVCHConfig(ctx, vch, sess)
-	v2.CheckFirewall(ctx)
-	firewallIssues := v2.GetIssues()
+	// System time
+	v.SystemTime = time.Now().Format(time.UnixDate)
 
-	if len(firewallIssues) == 0 {
-		v.FirewallStatus = GoodStatus
+	if sess == nil {
+		// We can't connect to vSphere
+		v.VCHReachable = false
+		v.VCHUnreachable = true
+		v.FirewallStatus = BadStatus
 		v.FirewallIssues = template.HTML("")
+
+		v.LicenseStatus = BadStatus
+		v.LicenseIssues = template.HTML("")
 	} else {
-		v.FirewallStatus = BadStatus
-		for _, err := range firewallIssues {
-			v.FirewallIssues = template.HTML(fmt.Sprintf("%s<span class=\"error-message\">%s</span>\n", v.FirewallIssues, err))
+
+		v.VCHReachable = true
+		v.VCHUnreachable = false
```

Comment:
yeah this hurts, see if you can figure out how to use an `else` in the template as you mentioned before.. or even `if` and then `if not` somehow :\